example/associations/many-to-many-with-associations.js

```diff
@@ -0,0 +1,86 @@
+var Promise, collections, connections, setupWaterline;
+
+Promise = require('bluebird');
+
+setupWaterline = Promise.promisify(require('../raw/bootstrap'));
+
+connections = {
+    associations: {
+        adapter: 'sails-orientdb',
+        host: 'localhost',
+        port: 2424,
+        user: 'root',
+        password: 'root',
+        database: 'example-waterline-manyToMany',
+        options: {
+            databaseType: 'graph',
+            unsafeDrop: 'true',
+            storage: 'memory'
+        }
+    }
+};
+
+collections = {
+    celebrity: {
+        migrate: 'drop',
+        tableName: 'celebrityTable',
+        identity: 'celebrity',
+        connection: 'associations',
+        attributes: {
+            name: 'string',
+            people: {
+                collection: 'person',
+                through: 'likes',
+                via: 'celebrity'
+            }
+        }
+    },
+    person: {
+        migrate: 'drop',
+        tableName: 'personTable',
+        identity: 'person',
+        connection: 'associations',
+        attributes: {
+            name: 'string',
+            celebrities: {
+                collection: 'celebrity',
+                through: 'likes',
+                via: 'person'
+            }
+        }
+    },
+    likes: {
+        migrate: 'drop',
+        tableName: 'likesTable',
+        identity: 'likes',
+        connection: 'associations',
+        attributes: {
+            since: 'date',
+            "in": {
+                model: 'person',
+                columnName: 'person'
+            },
+            out: {
+                model: 'celebrity',
+                columnName: 'celebrity'
+            }
+        }
+    }
+};
+
+setupWaterline({
+    collections: collections,
+    connections: connections
+}).then(function (ontology) {
+    return Promise.join(ontology.collections.celebrity.create({
+        name: 'Megan Fox'
+    }), ontology.collections.person.create({
+        name: 'John Smith'
+    }), function (celebrity, person) {
+        return ontology.collections.likes.create({
```

Comment:
if I change the `in` to `person` and `out` to `celebrity` it all works beautifully including the `populate()`! but the wrong way round so the edge is saying that Megan Fox is a fan of John Smith (which may well happen to be true, but not what I wanted :smile:)

I guess outside of a graph databases the joins don't necessarily have a direction(?), I haven't dug through the code, but is it simply because on the order of parameters that its drawing the edge in a given direction?